elements/rh-icon/rh-icon.ts

```diff
@@ -0,0 +1,174 @@
+import { LitElement, html, type PropertyValues, type TemplateResult } from 'lit';
+import { customElement } from 'lit/decorators/custom-element.js';
+import { property } from 'lit/decorators/property.js';
+import { state } from 'lit/decorators/state.js';
+import { classMap } from 'lit/directives/class-map.js';
+
+import { Logger } from '@patternfly/pfe-core/controllers/logger.js';
+import { InternalsController } from '@patternfly/pfe-core/controllers/internals-controller.js';
+
+import style from './rh-icon.css';
+
+type Renderable = unknown;
+
+export type IconResolverFunction = (set: string, icon: string) =>
+  Renderable | Promise<Renderable>;
+
+/**
+ * requestIdleCallback when available, requestAnimationFrame when not
+ * @param f callback
+ */
+const ric: typeof globalThis.requestIdleCallback =
+     globalThis.requestIdleCallback
+  ?? globalThis.requestAnimationFrame
+  ?? (async (f: () => void) => Promise.resolve().then(f));
+
+/** Fired when an icon fails to load */
+export class IconResolveError extends ErrorEvent {
+  constructor(
+    set: string,
+    icon: string,
+    /** The original error when importing the icon module */
+    public originalError: Error
+  ) {
+    super('error', { message: `Could not load icon "${icon}" from set "${set}".` });
+  }
+}
+
+/**
+ * Icons represents general concepts and can support text as a decorative element.
+ * The icon element, <rh-icon>, is a container that allows users to add icons of varying
+ * dimensions in the same area without shifting surrounding content.
+ * @summary Decorative element which supports related content
+ * @slot - Slotted content is used as a fallback in case the icon doesn't load
+ * @fires load - Fired when an icon is loaded and rendered
+ * @fires error - Fired when an icon fails to load
+ * @csspart fallback - Container for the fallback (i.e. slotted) content
+ * @cssprop --rh-icon-size - Override default icon size
+ */
+@customElement('rh-icon')
+export class RhIcon extends LitElement {
+  public static readonly styles: CSSStyleSheet[] = [style];
+
+  private static onIntersect: IntersectionObserverCallback = records =>
+    records.forEach(({ isIntersecting, target }) => {
+      const icon = target as RhIcon;
+      icon.#intersecting = isIntersecting;
+      ric(() => {
+        if (icon.#intersecting) {
+          icon.load();
+        }
+      });
+    });
+
+
+  private static io = new IntersectionObserver(RhIcon.onIntersect);
+
+  private static instances = new Set<RhIcon>();
+
+  private static resolve: IconResolverFunction = (set: string, icon: string): Renderable =>
+    import(`@rhds/icons/dist/${set}/${icon}.js`)
```

Comment:
should have caught this earlier, but `/dist/` should not be part of the icons export map

Reply:
I should have spoken up I questioned it myself, my fault. Would you like to resolve this before release?  Assuming we can remap with `exports` in icons repo?

```json
  "exports": {
    "./microns/*": "./dist/microns/*",
    "./standard/*": "./dist/standard/*",
    "./social/*": "./dist/social/*",
    "./ui/*": "./dist/ui/*"
  },
```

Reply:
might be preferable to just copy those files to the tarball root

Reply:
Opened a PR to fix: https://github.com/RedHat-UX/red-hat-icons/pull/9 PTAL